src/bootstrap/src/ferrocene/doc/certified_api_docs.rs

```diff
@@ -11,14 +11,18 @@ pub(crate) struct CertifiedApiDocs {
 
 impl Step for CertifiedApiDocs {
     type Output = PathBuf;
-    const DEFAULT: bool = false;
+    const DEFAULT: bool = true;
 
     fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
         run.alias("ferrocene-certified-api-docs")
     }
 
     fn make_run(run: RunConfig<'_>) {
-        run.builder.ensure(CertifiedApiDocs { target: run.target });
+        if run.target.try_certified_equivalent().is_some() {
+            run.builder.ensure(CertifiedApiDocs { target: run.target });
+        } else {
+            run.builder.info(&format!("No certified target for {}", run.target.to_string()));
+        }
```

Comment:
this is confusing if someone runs `x doc ferrocene-certified-api-docs` and nothing happens. let's make this a default condition instead.
```suggestion
    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
        run.alias("ferrocene-certified-api-docs").default_condition(run.target.try_certified_equivalent().is_some())
    }

    fn make_run(run: RunConfig<'_>) {
        run.builder.ensure(CertifiedApiDocs { target: run.target });
```

Reply:
`ShouldRun` does not have a `target` field. I was unable to find some alternative that might work for this:

```
error[E0609]: no field `target` on type `ShouldRun<'_>`
  --> src/bootstrap/src/ferrocene/doc/certified_api_docs.rs:17:73
   |
17 |         run.alias("ferrocene-certified-api-docs").default_condition(run.target.try_certified_equivalent().is_some())
   |                                                                         ^^^^^^ unknown field
   |
   = note: available field is: `builder`
```

 I did try to do similar. Did I perhaps miss something?

Reply:
ok. you can get a list of all targets that will be built with `builder.targets`. i suggest something like this:
1. in `should_run`, add a condition that at least one target is certified.
2. in `make_run`, if there's no certified target equivalent, check `builder.was_explicitly_invoked::<CertifiedApiDocs>`. if so, fail. if not, skip.

Reply:
the goal here is to avoid `x doc certified-api-docs` from simply doing nothing and exiting. that would really suck if bors passes and we discover later that we just didn't build the docs.

Reply:
actually, you can probably skip step 1 altogether if you implement step 2.